Added info and examples for new functionalities (#207)
replicatedhq/troubleshoot.sh: docs/source/analyze/node-resources.md

          - pass:
              message: This cluster has a node with enough memory.
```

> ### Filter by labels

Reviewer: ```suggestion
### Filter by labels

> Filtering by labels was introduced in Kots 1.19.0 and Troubleshoot 0.9.42.

```

          - pass:
              message: This cluster has a node with enough memory.
```

```yaml
    - nodeResources:

>         checkName: Must have at least 3 nodes with labels beta.kubernetes.io/os=linux and kubernetes.io/rol=database-primary-replica

Reviewer: Can you describe what these labels mean in plain English?
Also, is there a typo in `rol=`?

          - pass:
              message: This cluster has a node with enough memory.
```

```yaml
    - nodeResources:

>         checkName: Must have at least 3 nodes with 2 cores and labels kubernetes.io/hostname=docker-desktop and kubernetes.io/role=database-primary-replica

Reviewer: I wouldn't suggest putting the labels in the checkName; these should be human-readable. Maybe:

```suggestion
        checkName: Must have at least 3 nodes with 2 cores, with Docker Desktop and a database
```

Reviewer: I don't understand how this check actually works. Where's the check for the labels? Does it filter the nodes by ones with those labels such that count() < 3 means there must be 3 nodes with docker-desktop on them?

Author: The main Idea is to be sure that the nodes being analyzed have certain labels. Maybe the example I used was miss-leading, check the new one in the new commit. In this case, for a node to match the filter this conditions must be met:
- Have 2 cores
- Have a label "kubernetes.io/hostname"="docker-desktop"
- Have a label "kubernetes.io/role"="database-primary-replica"

Reviewer: OK, so it sounds like all filters get applied, so it means that we need 3 nodes that have those labels and minimum cores, etc. That makes sense, but frankly is a bit confusing. If you're going to go that route, you need to change the actual messages. Currently, someone could have 3 nodes, but you'll tell them they don't, just because they don't have 3 nodes **with** those labels. This might be better as a different example, or even separate examples.

Author: I've written a new, simpler example, only using the selector matchLabel (see last commit). Let me know what you think

        filters:
          cpuCapacity: "2"
          selector: 
            matchLabel: 
               kubernetes.io/hostname: docker-desktop
               kubernetes.io/role: database-primary-replica
        outcomes:
        - fail:
            when: "count() < 3"

>             message: This application requires at least 3 nodes.

Reviewer: ```suggestion
            message: This application requires at least 3 nodes, with 5 recommended.
```

          - pass:
              message: This cluster has a node with enough memory.
```
### Filter by labels

> Filtering by labels was introduced in Kots 1.19.0 and Troubleshoot 0.9.42.

Labels are intended to be used to specify identifying attributes of objects that are meaningful and relevant to users, but do not directly imply semantics to the core system. Labels can be used to organize and to select subsets of objects.

> Troubleshoot allows users to analyze nodes that match one or more labels. For example, to require a certain number of nodes with certain labels as a preflight check, e.g. to check if a node has the label ```kubernetes.io/role=database-primary-replica```, indicating that it is running the Mongo Operator. Further filters may be applied to the nodes matching the labels. 

Reviewer: ```suggestion
Troubleshoot allows users to analyze nodes that match one or more labels. For example, to require a certain number of nodes with certain labels as a preflight check. Multiple filters may be specified and all are required to match for the node to match.
```

```yaml
    - nodeResources:

>         checkName: Must have 1 node with 16 GB (available) memory and 5 cores (on a single node) running Mongo Operator (label kubernetes.io/role=database-primary-replica expected in node).

Reviewer: ```suggestion
        checkName: Must have at least 1 node running Mongo Operator (label kubernetes.io/role=database-primary-replica) with 16 GB (available) memory and 5 cores (on a single node).
```

Reviewer: Actually, the checkName should probably be simpler. Just Must have Mongo running. Sor something like that. The details could be in the fail message.

Reviewer: ```suggestion
        checkName: Must have Mongo running
```
